app/scripts/controllers/overview.js

```diff
@@ -1336,46 +1338,69 @@ function OverviewController($scope,
 
-      // Get the service class for this instance.
+      // Get the service class for this instance. Returns a promise.
       fetchServiceClass = function(instance) {
         var serviceClassName = ServiceInstancesService.getServiceClassNameForInstance(instance);
 
+        var serviceClass = _.get(state, ['serviceClasses', serviceClassName]);
+        if (serviceClass) {
+          return $q.when(serviceClass);
+        }
+
         // Check if we already have the service class or if a request is already in flight.
-        if (!_.has(state, ['serviceClasses', serviceClassName]) && !serviceClassPromises[serviceClassName]) {
+        if (!serviceClassPromises[serviceClassName]) {
           serviceClassPromises[serviceClassName] = DataService.get(serviceClassesVersion, serviceClassName, {}).then(function(serviceClass) {
             state.serviceClasses[serviceClassName] = serviceClass;
+            return serviceClass;
           }).finally(function() {
             delete servicePlanPromises[serviceClassName];
           });
         }
+
+        return servicePlanPromises[serviceClassName];
       };
 
-      // Get the service plan for this instance.
+      // Get the service plan for this instance. Returns a promise.
       fetchServicePlan = function(instance) {
         var servicePlanName = ServiceInstancesService.getServicePlanNameForInstance(instance);
 
         // Check if we already have the service plan or if a request is already in flight.
-        if (!_.has(state, ['servicePlans', servicePlanName]) && !servicePlanPromises[servicePlanName]) {
+        var servicePlan = _.get(state, ['servicePlans', servicePlanName]);
+        if (servicePlan) {
+          return $q.when(servicePlan);
+        }
+
+        if (!servicePlanPromises[servicePlanName]) {
           servicePlanPromises[servicePlanName] = DataService.get(servicePlansVersion, servicePlanName, {}).then(function(servicePlan) {
             state.servicePlans[servicePlanName] = servicePlan;
+            return servicePlan;
           }).finally(function() {
             delete servicePlanPromises[servicePlanName];
           });
         }
+
+        return servicePlanPromises[servicePlanName];
       };
 
       watches.push(DataService.watch(serviceInstancesVersion, context, function(serviceInstances) {
         state.serviceInstances = serviceInstances.by('metadata.name');
+
+        var promises = [];
         _.each(state.serviceInstances, function(instance) {
           var notifications = ResourceAlertsService.getServiceInstanceAlerts(instance);
           setNotifications(instance, notifications);
 
-          fetchServiceClass(instance);
-          fetchServicePlan(instance);
+          promises.push(fetchServiceClass(instance));
+          promises.push(fetchServicePlan(instance));
+        });
+
+        // Wait for all promises to complete before trying to sort the
+        // instances and check bindability.
+        $q.all(promises).then(function() {
```

Comment:
my concern with doing this is that as soon as one of them fails then it will reject the q.all and this will never fire.  i dont think thats really what we want.